reactor-netty-core/src/test/java/reactor/netty/resources/DefaultPooledConnectionProviderTest.java

```diff
@@ -465,6 +465,64 @@ void testDisposeInactivePoolsInBackground() throws Exception {
 		}
 	}
 
+	@Test
+	void testIssue1790FIFOPool() {
+		doTestIssue1790(true);
+	}
+
+	@Test
+	void testIssue1790LIFOPool() {
+		doTestIssue1790(false);
+	}
+
+	private void doTestIssue1790(boolean fifoPool) {
+		DefaultPooledConnectionProvider provider;
+		if (fifoPool) {
+			provider =
+			        (DefaultPooledConnectionProvider) ConnectionProvider.builder("testIssue1790")
+			                                                            .maxConnections(1)
+			                                                            .fifo()
+			                                                            .build();
+		}
+		else {
+			provider =
+			        (DefaultPooledConnectionProvider) ConnectionProvider.builder("testIssue1790")
+			                                                            .maxConnections(1)
+			                                                            .lifo()
+			                                                            .build();
+		}
+
+		DisposableServer disposableServer =
+				TcpServer.create()
+				         .port(0)
+				         .wiretap(true)
+				         .bindNow();
+
+		Connection connection = null;
+		try {
+			connection =
+					TcpClient.create(provider)
+					         .port(disposableServer.port())
+					         .wiretap(true)
+					         .connectNow();
+
+			assertThat(provider.channelPools).hasSize(1);
+
+			@SuppressWarnings({"unchecked", "rawtypes"})
+			InstrumentedPool<DefaultPooledConnectionProvider.PooledConnection> channelPool =
+					provider.channelPools.values().toArray(new InstrumentedPool[0])[0];
+			assertThat(channelPool.metrics()).isSameAs(channelPool);
```

Comment:
there is a lot of ceremony in this test. I would consider adding a comment, or better: changing the assertion's description or error message so that if it ever fails, it becomes clear this was asserting an assumption about reactor-pool's implementation (that the pool and the `metrics()` are one and the same)

Reply:
```suggestion
			assertThat(channelPool.metrics())
				.withFailMessage("Reactor-netty relies on Reactor-pool instrumented pool.metrics() to be the pool instance itself, got <%s> and <%s>", 
				channelPool.metrics(), channelPool)
				.isSameAs(channelPool);
```